pkg/operation/operation.go

```diff
@@ -116,13 +116,51 @@ func newOperation(
 	}
 
 	if shoot != nil {
-		shootObj, err := shootpkg.New(k8sGardenClient, k8sGardenCoreInformers, shoot, gardenObj.Project.Name, disableDNS, gardenObj.InternalDomain, gardenObj.DefaultDomains)
+
+		binding, err := k8sGardenCoreInformers.SecretBindings().Lister().SecretBindings(shoot.Namespace).Get(shoot.Spec.SecretBindingName)
```

Comment:
I think the responsibility of `shootpkg.New` was clearer before this change because the function almost calculated everything required for the shoot object. Now most of the complexity is outside while there is still a certain portion in `shootpkg.New`, i.e. I don't see a clear separation.
WDYT or what was the motivation?

Reply:
I wanted to make it more testable, I'm slowly start to refactor big portions of the code and to make it more isolated from each other. 
With this change everything which the `Shoot` needs is passed and it doesn't actually need to perform any network calls to populate it's fields.

Reply:
For example `k8sGardenClient`, `k8sGardenCoreInformers` were only there to so the Shoot can fetch the `CloudProfile`, get extensions and so - this data can (and probably should) be passed from the top.

Reply:
> I wanted to make it more testable, I'm slowly start to refactor big portions of the code and to make it more isolated from each other.

That's nice and I understand that but on the other hand this seems like only shifting complexity (here from `shootpkg.New` to `operation.New`) at the moment. Actually, `shootpkg.New` is already testable but of course mock and lister preparation is needed for that.

Instead of a better testability, I rather see a separation of concern being broken here. 
Maybe it'd be more obvious to me if you had more refactoring in this area.

Reply:
+ Testing `shootpkg.New` as it is today will enable a lower risk refactoring in the future.

Reply:
Yes, I'm moving complexity (or lets say dependency injection) to higher in the stack. With this PR, no mocks are needed are needed.

Reply:
Instead of moving the stuff to the `operation.New`, WDYT about moving it to a separate function in the `shootpkg`? You could make `New` as you have it now (just pass the already read/computed objects), and in another function `NewXYZ` function you read/compute them and call `New` afterwards?

Reply:
> Instead of moving the stuff to the operation.New, WDYT about moving it to a separate function in the shootpkg?

Why? The `Shoot` has a specific dependencies and it should have only way to construct it. Having multiple `NewXYZ` is pointless..

Reply:
Also, I haven't looked at it fully, but I have suspicion that higher in the stack `cloudprofile`, `seedName` and other similar resources are already available and simply not passed to here

Reply:
> it should have only one way to construct it. Having multiple NewXYZ is pointless..

Can you elaborate? I don't get the rationale behind the "should" and "pointless" words.

What I'm proposing is just to bundle the read/compute stuff into  "helper function" (that is called `NewXYZ`) instead of doing it in the `operation.New` function. It will not only make the `operation.New` function easier, it will also help others that want to instantiate the `Shoot` object by enabling reuse. ?

Reply:
> Also, I haven't looked at it fully, but I have suspicion that higher in the stack cloudprofile, seedName and other similar resources are already available and simply not passed to here

+1 but this is why I think it'd make sense to include this change in a "broader" refactoring where we can see that it makes sense to move certain complexity/DIs to other places.